dali/python/nvidia/dali/types.pyi

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+
+class DALIDataType(Enum):
+    NO_TYPE = ...
+    UINT8 = ...
+    UINT16 = ...
+    UINT32 = ...
+    UINT64 = ...
+    INT8 = ...
+    INT16 = ...
+    INT32 = ...
+    INT64 = ...
+    FLOAT16 = ...
+    FLOAT = ...
+    FLOAT64 = ...
+    BOOL = ...
+    STRING = ...
+
+class DALIImageType(Enum):
+    RGB = ...
+    BGR = ...
+    GRAY = ...
+    YCbCr = ...
+    ANY_DATA = ...
+
+class DALIInterpType(Enum):
+    INTERP_NN = ...
+    INTERP_LINEAR = ...
+    INTERP_CUBIC = ...
+    INTERP_LANCZOS3 = ...
+    INTERP_TRIANGULAR = ...
+    INTERP_GAUSSIAN = ...
```

Comment:
Is that written manually? Do we have at least some way to ensure that it doesn't go stale?

Reply:
It is, and I don't have a good idea how to check it, considering the fact, that we already have it spelled out in this or that form in 5 places.
The only thing for my defense is that we didn't change it for quite a long time, and this definition is not binding. 
The problem with those enums is, that they are only visible when the actual pybind module is loaded, so any static tool that looks just at the Python code, doesn't see the original definitions. This was the only way, that I found, not to pollute the `fn` module (and submodules) with the `class DALIInterpType: ...` placeholders, that were visible in the intellisense. Otherwise we would need to rewrite those types in Python, maybe this should be the way.